app/src/main/java/com/github/weg_li_android/PhotoRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,161 @@
+    private val mInflater: LayoutInflater = LayoutInflater.from(context)
+    private var mClickListener: ItemClickListener? = null
+    var deletePhotoSet: ObservableArrayList<Int> = ObservableArrayList()
+    /*init {
+        deletePhotoSet.addOnListChangedCallback(
+            RecyclerViewAdapterOnListChangedCallback<ObservableList<Int>>(deletePhotoSet)
+        )
+    }*/
+
+    @NonNull
+    override fun onCreateViewHolder(
+        @NonNull parent: ViewGroup,
+        viewType: Int
+    ): ViewHolder {
+        val view: View = mInflater.inflate(R.layout.photo_list_item, parent, false)
+        view.photo_selected_check.visibility = View.VISIBLE
+        return ViewHolder(view)
+    }
+
+    // binds the data to the TextView in each cell
+    override fun onBindViewHolder(
+        @NonNull holder: ViewHolder,
+        position: Int
+    ) {
+        val image : Bitmap? = getThumbnail(data[position])
+        holder.myAppCompatImageView.background = image?.toDrawable(resources)
+        if(position in deletePhotoSet) {
+            holder.selectedCheckAppCompatImageView.visibility = View.VISIBLE
+        }
+        else holder.selectedCheckAppCompatImageView.visibility = View.GONE
+    }
+/*
+    private class RecyclerViewAdapterOnListChangedCallback<Int>(myAppCompatImageView: ObservableArrayList<kotlin.Int>) : ObservableList.OnListChangedCallback<ObservableList<Int>>() {
+        val mAppCompatImageView = myAppCompatImageView
+        override fun onChanged(sender: ObservableList<Int>?) {
+
+            Timber.e("yooo changed!")
+            //adapter.notifyDataSetChanged()
+        }
+
+        override fun onItemRangeRemoved(sender: ObservableList<Int>?, positionStart: kotlin.Int, itemCount: kotlin.Int) {
+            Timber.e("yooo range removed!")
+            //adapter.notifyItemRangeRemoved(positionStart, itemCount)
+        }
+
+        override fun onItemRangeMoved(sender: ObservableList<Int>?, fromPosition: kotlin.Int, toPosition: kotlin.Int,  itemCount: kotlin.Int) {
+            Timber.e("yooo range mooved!")
+        }
+
+        override fun onItemRangeInserted(sender: ObservableList<Int>?, positionStart: kotlin.Int, itemCount: kotlin.Int) {
+            Timber.e("yooo range inserted!")
+            Timber.e("This is the new list"+sender.toString())
+            //adapter.notifyItemRangeInserted(positionStart, itemCount)
+        }
+
+        override fun onItemRangeChanged(sender: ObservableList<Int>?, positionStart: kotlin.Int, itemCount: kotlin.Int) {
+            Timber.e("yooo range changed!")
+            //adapter.notifyItemRangeChanged(positionStart, itemCount)
+        }
+
+    }
+*/
+    @Throws(FileNotFoundException::class, IOException::class)
+    fun getThumbnail(uri: Uri?): Bitmap? {
+        if(context !=null) {
+            val thumbnailSize = 120f.convertDpToPixel(context).toInt()
+            val thumbnail: Bitmap? =
+                uri?.let {
+                    context.contentResolver.loadThumbnail(
```

Comment:
This is an API29 method, but our min sdk level is 19 or 21. It will always crash. Android Studio should also give you an error for this.